packages/berry-pnpify/sources/NodeModulesFS.ts

```diff
@@ -0,0 +1,266 @@
+  }
+
+  accessSync(p: string, mode?: number) {
+    return this.baseFs.accessSync(this.resolveStatPath(p), mode);
+  }
+
+  async statPromise(p: string) {
+    return await this.baseFs.statPromise(this.resolveStatPath(p));
+  }
+
+  statSync(p: string) {
+    return this.baseFs.statSync(this.resolveStatPath(p));
+  }
+
+  async lstatPromise(p: string) {
+    return await this.baseFs.lstatPromise(this.resolveStatPath(p));
+  }
+
+  lstatSync(p: string) {
+    return this.baseFs.lstatSync(this.resolveStatPath(p));
+  }
+
+  async chmodPromise(p: string, mask: number) {
+    return await this.baseFs.chmodPromise(this.throwIfPathReadonly(p), mask);
+  }
+
+  chmodSync(p: string, mask: number) {
+    return this.baseFs.chmodSync(this.throwIfPathReadonly(p), mask);
+  }
+
+  async renamePromise(oldP: string, newP: string) {
+    return await this.baseFs.renamePromise(this.throwIfPathReadonly(oldP), this.throwIfPathReadonly(newP));
+  }
+
+  renameSync(oldP: string, newP: string) {
+    return this.baseFs.renameSync(this.throwIfPathReadonly(oldP), this.throwIfPathReadonly(newP));
+  }
+
+  async copyFilePromise(sourceP: string, destP: string, flags?: number) {
+    return await this.baseFs.copyFilePromise(sourceP, this.throwIfPathReadonly(destP), flags);
+  }
+
+  copyFileSync(sourceP: string, destP: string, flags?: number) {
+    return this.baseFs.copyFileSync(sourceP, this.throwIfPathReadonly(destP), flags);
+  }
+
+  async writeFilePromise(p: string, content: string | Buffer | ArrayBuffer | DataView, opts?: WriteFileOptions) {
+    return await this.baseFs.writeFilePromise(this.throwIfPathReadonly(p), content, opts);
+  }
+
+  writeFileSync(p: string, content: string | Buffer | ArrayBuffer | DataView, opts?: WriteFileOptions) {
+    return this.baseFs.writeFileSync(this.throwIfPathReadonly(p), content, opts);
+  }
+
+  async unlinkPromise(p: string) {
+    return await this.baseFs.unlinkPromise(this.throwIfPathReadonly(p));
+  }
+
+  unlinkSync(p: string) {
+    return this.baseFs.unlinkSync(this.throwIfPathReadonly(p));
+  }
+
+  async utimesPromise(p: string, atime: Date | string | number, mtime: Date | string | number) {
+    return await this.baseFs.utimesPromise(this.resolveStatPath(p), atime, mtime);
+  }
+
+  utimesSync(p: string, atime: Date | string | number, mtime: Date | string | number) {
+    return this.baseFs.utimesSync(this.resolveStatPath(p), atime, mtime);
+  }
+
+  async mkdirPromise(p: string) {
+    return await this.baseFs.mkdirPromise(this.throwIfPathReadonly(p));
+  }
+
+  mkdirSync(p: string) {
+    return this.baseFs.mkdirSync(this.throwIfPathReadonly(p));
+  }
+
+  async rmdirPromise(p: string) {
+    return await this.baseFs.rmdirPromise(this.throwIfPathReadonly(p));
+  }
+
+  rmdirSync(p: string) {
+    return this.baseFs.rmdirSync(this.throwIfPathReadonly(p));
+  }
+
+  async symlinkPromise(target: string, p: string) {
+    return await this.baseFs.symlinkPromise(this.throwIfPathReadonly(target), p);
```

Comment:
I think `throwIfPathReadonly` should be called on `p`, not `target` (`target` is the "content" of the symlink at location `p`)